docs/retention_policy.md

```diff
@@ -0,0 +1,93 @@
+
+# Checkpoint Retention Policy Documentation
+
+## Overview
+
+The checkpoint retention policy in the CheckpointRestoreOperator allows users to manage and configure how checkpoints are retained and cleaned up within a Kubernetes cluster. This document provides an overview of how to configure these policies, the hierarchy of their application, and details about specific fields.
+
+## Applying a Retention Policy
+
+To apply a retention policy, you need to create a `CheckpointRestoreOperator` resource. Below is an example configuration:
+```yaml
+`apiVersion: criu.org/v1
+kind: CheckpointRestoreOperator
+metadata:
+  labels:
+    app.kubernetes.io/name: checkpointrestoreoperator
+    app.kubernetes.io/instance: checkpointrestoreoperator-sample
+    app.kubernetes.io/part-of: checkpoint-restore-operator
+    app.kubernetes.io/managed-by: kustomize
+    app.kubernetes.io/created-by: checkpoint-restore-operator
+  name: checkpointrestoreoperator-sample
+spec:
+  checkpointDirectory: /var/lib/kubelet/checkpoints
+  applyPoliciesImmediately: false
+  globalPolicy:
+    maxCheckpointsPerNamespace: 50
+    maxCheckpointsPerPod: 30
+    maxCheckpointsPerContainer: 10
+  # containerPolicies:
+  #   - namespace: namespace
+  #     pod: pod_name
+  #     container: container_name
+  #     maxCheckpoints: 5
+  # podPolicies:
+  #   - namespace: namespace
+  #     pod: pod_name
+  #     maxCheckpoints: 10
+  # namespacePolicies:
+  #   - namespace: namespace
+  #     maxCheckpoints: 15` 
```

Comment:
I think we can uncomment this since the example is only meant to show the different ways it can be configured.